content/blogs/2024-04-20-top-10-cool-features-I-love-about-kestra.md

```diff
@@ -0,0 +1,150 @@
+
+## 4. Inputs
+
+Kestra flows can be configured to accept different types of [inputs](https://kestra.io/docs/workflow-components/inputs) like string, integer, float, date, datetime, and many more. My favorite one is the FILE input type.
+
+For many orchestration tools, it is a tedious task to provide file input, as the file either needs to be checked in to the code or uploaded to an apporopriate loaction in the file system. With Kestra's file input, you can literally choose file from any location of your machine, and provide it as an input. This makes it a piece of cake to try out the workflows with different test files. It could not be any simpler to make changes to the file and iterate the testing process.
+
+![input_file](/blogs/2024-04-20-top-10-cool-features-I-love-about-kestra/input_file.png)
+
+## 3. Secrets
+
+It is paramount to prioritize the protection of sensitive information. It is a necessity that no sensitive information gets checked-in in its plain or easily-decodeable format. At the same time, it should also be easy to introduce this information in the platform for developer productivity.
+
+Kestra just has the right solution in place. The [secrets](https://kestra.io/docs/concepts/secret) can be provided as environment variables in base64 encoded format while operating Kestra in the dockerized mode. The secrets can also be provided from the UI directly in its EE edition. Using secret in Kestra is pretty starightforward as well. You can use the format `{{ secret('MY_PASSWORD') }}` and access a secret stored under `SECRET_MY_PASSWORD` environment variable. Overall, its mesmerizing to see how easy it is to introduce and use secrets in Kestra.
+
+Here is an image of adding secret via the UI in EE edition:
+
+![secrets_ee](/blogs/2024-04-20-top-10-cool-features-I-love-about-kestra/secrets_ee.png)
+
+## 2. Task defaults
+
+This is yet another powerful feature from the developer productivity perspective. Generally, you develop a pipeline in correspondance to some technology, and it is extremely likely that you use multiple tasks that corresponds to the same technology. For example, in a flow that queries Redshift, it is very likely that you connect to Redshift for creating the table in one task, and then to insert data into it in another task, and then query it for some purpose. In this case, you would just end up duplicating the Redshift connection information in all these tasks. This hampers the developer productivity, and also leads to config duplication.
+
+In order to avoid this duplication, we have Kestra [task defaults](https://kestra.io/docs/workflow-components/task-defaults) to the rescue. Mention the task defaults once in the flow, and it gets referenced in all the tasks of the corresponding type.
```

Comment:
```suggestion
In order to avoid this duplication, Kestra provides [task defaults](https://kestra.io/docs/workflow-components/task-defaults). Mention the task defaults once in the flow, and it gets referenced in all the tasks of the corresponding type.
```